File: checks/ca.py
import os; import sys; sys.path.insert(1, os.path.join(sys.path[0], '..'))

import pandas as pd
import prince
from sklearn import datasets


X = pd.read_csv('children.csv', index_col=0)

ca = prince.CA().fit(X)

print('Eigenvalues')
print(ca.eigenvalues_)
print(ca.explained_inertia_)
print('---')

print('U')
print(ca.U_[:5])
print('---')

print('V')
print(ca.V_)
print('---')

print('s')
print(ca.s_)
print('---')

print('Row coords')
print(ca.row_coordinates(X)[:5])
print('---')

File: checks/ca.R
library(FactoMineR)

data(children)

X <- children[c(1:5)]

ca <- CA(X, graph=FALSE)

print(ca$eig)
print("---")

print("U")
print(ca$svd$U[1:5,])
print("---")

print("V")
print(ca$svd$V)
print("---")

print("s")
print(ca$svd$vs)
print("---")

print("Row coords")
print(ca$row$coord[1:5,])
print("---")
